client/packages.go

```diff
@@ -40,6 +40,8 @@ type Snap struct {
 	Type          string    `json:"type"`
 	Version       string    `json:"version"`
 	Revision      Revision  `json:"revision"`
+	Confinement   string    `json:"confinement"`
```

Comment:
Should we have a similar set of consts to the ones we have under snap/? `StrictConfinment`, etc.

Also, we should probably add a `DevMode` flag too, which indicates what the current status of the snap is. There's a distinction between the opinion of the snap about how it should be handled, and how the system is actually handling it. One is more useful when considering the snap for installation, while the other is more useful when we want to know what's the installed status of the snap in the system.